### DISCUSSION TITLE
[Deprecated] Uploading MP3 Files to Waze's Custom Voice Navigation Feature Without a Smartphone (2024)

### DISCUSSION THREAD

**Discussion by pipeeeeees** [pipeeeeees/waze-voicepack-links#31] [General]
**NOTE**: This method is no longer recommended. See https://github.com/pipeeeeees/waze-voicepack-links/tree/main/mp3_upload#how-to-upload-your-own-mp3-files-to-make-a-waze-voicepack-link instead

Uploading mp3 files to Waze has been attempted for well over a decade now. For me personally, it started with trying to get Morgan Freeman's voice back on my iPhone 5 in 2014 by jailbreaking the device and using ifunbox to swap audio files. It worked, but times have changed. Jailbreaking is harder these days. Nobody wants to go through the hassle. Plus new features such as the now mature Custom Voice Directions in Waze has made sharing voices extremely easy with the share of a link. There are some limitations though. Namely:

1. you cannot upload mp3 files in the mobile app. To do it the normal way, it must be recorded with your mobile device microphone (which could be crappy). 
2. the app also heavily compresses the audio file which makes playback sound muffled.

Old tutorials are outdated and no longer work. There are a slew of people over the years who have attempted to get it working and do, but lose passion over the years and go internet quiet (I'm starting to become that way with my involvement with the Waze voicepacks area)... then no longer work and the cycle continues... so I decided to create this internet discussion post for those present and future to read and learn from. Without further ado, here is how you can upload MP3 files onto Waze without a smartphone

# What you will need

1. a computer with OS capable of running an android emulator (so at least a dual core processor, 8GB ram but more is recommended, an OK GPU)
2. internet connection
3. patience

# Overview

The process is as follows: 
1. Download an android emulator on the PC that is capable of rooting. Bluestacks 5 (NOT 10) is capable of this and is what I use. 
2. Root the android emulation to grant you access to the root files. 
3. We will want to download an app from the playstore in the emulation to allow us to navigate the root file system (I use the solid explorer app... there is a free trial and I paid for it after... use whatever root fs you want for this as long as you can copy paste files). 
4. Download the waze app in the emulation and make a new custom voice (no need to log in, and best if you do not - you can screw up your personal account in unexpected ways I found). 
5. Fill every voice line with like 1 second of audio. I found it helpful to make a macro to auto record 1 second audio for all phrases just because ive done this hundreds of times. This step can be mindless and time consuming otherwise. After recording every phrase with some short audio, SAVE and name the pack to what you want. (This uploads valid audio files to Waze's servers and "makes room" as I call it). DO NOT FORGET: if you want the voice to have both miles and meters unit directions available, you need to save and go to settings and change the metric from meters to miles / vice versa and go back to record 1 second of audio for miles or meters.
6. Next (and this is important) open back up your custom voice pack for editing/recording and leave it running in the background. This menu gives the mp3 files write access for you to swap the mp3 files so it must stay running in the background on the editing menu. Otherwise it won't work
7. while step 6 is running in the background, switch to your root file system app and go to the following directory: root/data/user/0/com.waze/waze/custom_prompts_temp
8. in that directory are all the 1 second audio files we just recorded and will want to replace. with write access enabled in step 6, go ahead and use your root file system explorer app to replace those mp3 files with the desired mp3 files you want (and only those mp3 files matched with exact same naming. if you add extras, it will likely not work... for instance if you did not record 1 second audio for the 200.mp3 or any other mile mp3 file and then add it in without the 1 second audio files first being present on Waze's servers, the waze app will catch the discrepancy and reject your changes.. from my experience). A full list of all phrase file names are [here](https://github.com/pipeeeeees/waze-voicepack-links/blob/main/conversion/mp3_filenames.txt).
9. go back to the waze app which should have had the edit screen up in the background and just scroll from top to bottom to top again. I think this step is important as updating the screen elements while scrolling seems to save the files during write access, but I do not remember if this really matters or not. Just do it if you're not trying to start back to step 5 again... but you may be ok not scrolling top to bottom to top.
10. save the voicepack.
11. you should be able to share the pack now. 
- If the share button goes gray and you cannot share the pack OR if you can share it but only the 1 second audio clips are playable from another device using the share link, then you have screwed up the upload to Waze's servers. They rejected your new mp3 files. The most common reason for this is the collective filesize of all mp3 files is too large. 0.8MB is the limit their servers will accept for the aggregate filesize of all mp3 files in a custom voicepack. If all of the uploaded files exceed this, usually their server rejects your modified mp3 files and the share button will go gray right after saving. The share link may turn blue when leaving and that menu and returning and sharing the link, but you will find the voicelines do not download to anyone else using the line. 
- This is why I made this python script to optimize the filesizes of each mp3 to maximize audio quality while staying under that 0.8MB aggregate filesize limit: https://github.com/pipeeeeees/waze-voicepack-links/blob/main/conversion/converter.py - other ways for less python/code savvy users is to reduce the bitrate of each audio file (try limiting to 50k bitrate... anything below 32k becomes hard to listen to) or shortening the long audio files with a computer app like Audacity (there is no limit on length of mp3s per voice phrase that I found... but longer mp3's have a much larger filesize)

Important thing to keep in mind is Waze offers this Custom Voice Direction feature FOR FREE. So please do not abuse this. Storing all of our audio files to their servers is not free for them... this is why they compress and make the audio quality so bad to save storage space and save money. If they were to get upset, they have fiscal justification already to axe this feature. They have a track record of making this process of uploading mp3 files backdoor style harder already, so yeah. Be appreciative, fellow wazers!

**Comment by Bigorno56** [comment]
Thanks for the tutorial. I managed to make a CARS voice pack in French.

**Comment by bepis-music** [comment]
unfortunately i've hit a dead end at step 4... the app can't be downloaded from the play store from within the emulator (not compatible) and a manually downloaded APK doesn't run. bluestacks 5 comes running android 7... is there any way to change that?

**Reply by pipeeeeees** [reply]
I am not sure. It has been while since I set one up from scratch. I thought selecting between a few android versions was possible, but maybe not?

Could try a different emulator, I know at least one person who was successful without using bluestacks specifically

**Reply by bepis-music** [reply]
Got it to work with Memu emulator, except when i "tap to download" the finished custom voice pack thing, it just downloads for a fraction of a second then goes back to "tap to download." I may have done something wrong, or maybe they patched that little glitch we found. I just want miku to tell me where to go on my phone lmao why is this so hard man
(I also did try emailing the voice pack to myself. here's the link if you wanna see if you can figure out what's wrong: https://waze.com/ul?acvp=e573f395-d98f-4341-9ba5-bb0424558a7e )

**Reply by pipeeeeees** [reply]
> Got it to work with Memu emulator, except when i "tap to download" the finished custom voice pack thing, it just downloads for a fraction of a second then goes back to "tap to download." I may have done something wrong, or maybe they patched that little glitch we found. I just want miku to tell me where to go on my phone lmao why is this so hard man
> 
> (I also did try emailing the voice pack to myself. here's the link if you wanna see if you can figure out what's wrong: https://waze.com/ul?acvp=e573f395-d98f-4341-9ba5-bb0424558a7e )

Not sure what you're trying to do here if it's already a voice pack?

**Reply by bepis-music** [reply]
it's not a usable voice pack from what I can tell. It doesn't actually download; I think I did something wrong, but I'm not sure what. It just gets stuck on tap to download--when I tap it it does a loading thing for a fraction of a second then goes right back to saying tap to download.

**Reply by pipeeeeees** [reply]
Ah I see. Yeah it's broken or something. Interesting

**Reply by juangui61gar-oss** [reply]
Pipee, can you make the Avengers pack? I have the audio files but I wasn't able to upload them, based on your explanation. I can send you the audio files; this is my email if you can help me: sirlawyer1@gmail.com. Thanks!

**Reply by juangui61gar-oss** [reply]
Pipee, can you make the Avengers pack? I have the audio files but I wasn't able to upload them, based on your explanation. I can send you the audio files; this is my email if you can help me: sirlawyer1@gmail.com. Thanks!

**Comment by roymustang197** [comment]
Hey, is there a way to extract an official voice? Master Chief (Spanish) came back and I want to download it but I cant find the root folder with the files

**Reply by roymustang197** [reply]
Nevermind I just found it here: root/data/user/0/com.waze/waze/sound/XXXX

**Reply by SomePers-s-on** [reply]
Hiya! I was going to do the same thing*, but I cannot find the voice I am looking for in root/data/user/0/com.waze/waze/sound .
 Would you tell me what you see when you go to root/data/user/0/com.waze/waze/sound and what apps you used? 
Help would be very much appreciated. :)

*extracting the Venom voice files to be able to use them in the future when licensing for them may run out, to be specific.

**Comment by kaylat15** [comment]
If you can't do the python method because you don't know anything about python, this is for you. If you are having issues like the voice not downloading or the share is grayed out all you need to do is put them in this online audio compressor and set the constant bitrate to 48 kbps then replace the files again while your voice is on the edit menu. https://www.xconvert.com/audio-compressor

**Reply by pipeeeeees** [reply]
Hey thanks for sharing! 

Right if it's all greyed out, the files are collectively over the limit of 0.8MB. 48kbps is a great rule of thumb, but may leave some audio quality on the table or may not be enough compression. Sometimes packs made with really long mp3 files can be compressed to 48kbps and still be over the limit. So ymmv. 

In the past I've tried compressing some of the more rare lines to 32kbps (like roundabout exit 6 and 7) while leaving the others at, say, 48kbps to save on space and maximize quality. However, that's really pinching pennies. I've found cutting many of the StartDrive mp3s in half or shorter really helps reduce filesize and keep bitrate high. 

So hence my python script conversion.py that does its best to optimize bitrate. However, hopefully those tips in the previous paragraph can help anyone trying to maximize quality and avoid grey share buttons!

**Reply by juangui61gar-oss** [reply]
Pipee, can you make the Avengers pack? I have the audio files but I wasn't able to upload them, based on your explanation. I can send you the audio files; this is my email if you can help me: sirlawyer1@gmail.com. Thanks!

**Comment by nobigdeal13** [comment]
Apparently this doesn't work with LDPlayer, spent quite a bit of my time trying to get it to work since it's one of the easier emulator to "root". At this point I'm sure any other emulators that have some sort of a "enable root" option in the emulator setting is not going to work. Bluestacks with Kitsune Mask works, meaning that it must be a rooting process similar to any other phones and not using any external settings like the emulator's setting. Just a heads up for others planning to create their own packs.

**Comment by vshopov996** [comment]
I appreciate all the effort that went into figuring this out. Took me about 5 hrs even with this guide but I finally made me own pack, thanks.
Also if anyone wants to do this BlueStacks5 didn't work for me. Whenever I tried to download Waze it would say it's not compatible with my device and I would get an error when I tried downloading newer android versions. I used MEmu instead. Also used FV File Explorer because I was getting access denied when trying to access the root folders in Solid Explorer.

**Comment by Ryuunosuk3** [comment]
Hey, I'm embarrassed that this is the issue I'm having, but I haven't been able to find anything out on the internet. Does anyone know how to open back up an already recorded Waze voice? It doesn't seem to let me.. Many thanks!

**Reply by pipeeeeees** [reply]
Yes, take the link (for example: https://waze.com/ul?acvp=6086eb80-ee87-4841-93ae-1f212817293d) and modify it to look like this https://voice-prompts-ipv6.waze.com/6086eb80-ee87-4841-93ae-1f212817293d.tar.gz and bobs your uncle it should download a compressed file that when uncompressed contains the mp3 files as a backup

**Comment by hifny2000-sudo** [comment]
hello...
no option to share your custom voice available now in waze..
anyone know what to do

**Reply by pipeeeeees** [reply]
not sure what you mean by this

**Comment by purwantostretton-source** [comment]
com.wave isnt showing up in the root explorer. im using an older version of waze that i installed through an apk btw.

**Reply by purwantostretton-source** [reply]
i found the the folder in /data_mirror/data_ce/null/0/com.waze/waze/custom_prompts_temp 
just incase anyone else has this same problem

**Comment by lejon0336-gif** [comment]
Yeah, now i'll make Kaatje's voice

**Comment by dylanbrookovergames** [comment]
i literally cannot do this in 2026. bluestacks cannot be rooted. It says it is, but it literally isn't cause i can't open the /root folder and all that. edited all the text files and all

**Reply by kaylat15** [reply]
The new bluestacks x doesn't work to root. You need to use bluestacks 5

**Reply by dylanbrookovergames** [reply]
Yes, i was using bluestacks 5. made sure to downgrade to the 5.22 ones too. even if you do it on the latest blue5 you cant root it

**Comment by Bypus** [comment]
Hello, I've tried this and all sounds work except the distances.
I only put distances in meters and they are all ~1.2 second long.
I have tried going up and down, saving, reload the app but the distances are only silence (my placeholder)

**Comment by juangui61gar-oss** [comment]
Pipee, can you make the Avengers pack? I have the audio files but I wasn't able to upload them, based on your explanation. I can send you the audio files; this is my email if you can help me: sirlawyer1@gmail.com. Thanks!
